src/xnnpack/intrinsics-polyfill.h

```diff
@@ -407,17 +407,57 @@ float Q6_f32_vrsum_Vsf(HVX_Vector vin){
     return *((float *) &vin);
 }
 
-// Temporary div implementation for HVX.
-// TODO: improve the implementation 
-//       e.g., use reprocical with newton-raphson approximation
+// DIV implementation using Newton-Raphson reciprocal approximation
+// Implementation comes from Halide project
+// a/b = a * fast_inverse__vsf(b)
 static XNN_INTRINSIC
-HVX_Vector Q6_Vsf_vdiv_VsfVsf(HVX_Vector vin1, HVX_Vector vin2){
-    float* svin1 = (float *) &vin1;
-    float* svin2 = (float *) &vin2;
+HVX_Vector fast_inverse__vsf(HVX_Vector vin) {
+    const uint32_t fp_exp_norm = 0x7F000000;  // IEEE sf: sign=0, exp=254, mant=0
+    const uint32_t fp_exp_mask = 0xFF800000;  // mask for IEEE sf exp
+    const uint32_t nr_T1       = 0x5a5a5a7f;  // Newton Raphson T1=24.0/17.0 (qf32)
+    const uint32_t nr_T2       = 0x8787877d;  // Newton Raphson T2=-8.0/17.0 (qf32)
+    const uint32_t qf_one      = 0x4000007F;  // 1.0 (qf32)
+
+    HVX_Vector vfp_exp_norm = Q6_V_vsplat_R(fp_exp_norm);
+    HVX_Vector vfp_exp_mask = Q6_V_vsplat_R(fp_exp_mask);
+
+    HVX_Vector vnr_T1 = Q6_V_vsplat_R(nr_T1);
+    HVX_Vector vnr_T2 = Q6_V_vsplat_R(nr_T2);
+
+    HVX_Vector vone  = Q6_V_vsplat_R(qf_one);
+    HVX_Vector vzero = Q6_V_vzero();
+
+    // IEEE sf: sign[i] = sign(den[i]), exp[i] = exp(den[i]), mant = 0
+    HVX_Vector vfp_exp = Q6_V_vand_VV(vin, vfp_exp_mask);
+
+    // normalization factor in IEEE sf:
+    //   sign[i] = sign(den[i]), exp[i] = 254 - exp(den[i]), mant = 0
+    HVX_Vector vfp_norm = Q6_Vw_vsub_VwVw(vfp_exp_norm, vfp_exp);
+    HVX_Vector vnorm = Q6_Vqf32_vadd_VsfVsf(vfp_norm, vzero);  // qf32
+
+    HVX_Vector vout = Q6_Vqf32_vmpy_VsfVsf(vin, vfp_norm);  // normalize den[i]
+
+    // initial estimate X0[i] = T1 + (T2 * den[i])
+    HVX_Vector vtmp = Q6_Vqf32_vmpy_Vqf32Vqf32(vnr_T2, vout);
+    HVX_Vector vX0  = Q6_Vqf32_vadd_Vqf32Vqf32(vnr_T1, vtmp);
+
+#pragma clang loop unroll(enable)
```

Comment:
I've had problems with this pragma on arm and suggest manually unrolling, or dont worry about it.
recent clang with -O2 is super aggressive on unrolling, and needs discouragement if anything :-)